ports/servoshell/desktop/minibrowser.rs

```diff
@@ -267,30 +348,27 @@ impl Minibrowser {
 
             let mut embedder_events = vec![];
 
-            // A simple Tab header strip, using egui 'SelectableLabel' elements.
-            // TODO: Add a way to close a tab eg. with a [x] control.
+            // A simple Tab header strip
             TopBottomPanel::top("tabs").show(ctx, |ui| {
                 ui.allocate_ui_with_layout(
                     ui.available_size(),
                     egui::Layout::left_to_right(egui::Align::Center),
                     |ui| {
                         for (webview_id, webview) in webviews.webviews().into_iter() {
-                            let msg = match (&webview.title, &webview.url) {
-                                (Some(title), _) if !title.is_empty() => title.clone(),
-                                (_, Some(url)) => url.to_string(),
-                                _ => "New Tab".to_owned(),
+                            let label = match (&webview.title, &webview.url) {
+                                (Some(title), _) => title,
+                                (None, Some(url)) => &url.to_string(),
+                                _ => "New Tab",
```

Comment:
This could avoid the match, I think:

```suggestion
                            let label = webview.title
                                .or(webview.url.map(ServoUrl::to_string))
                                .as_ref()
                                .unwrap_or_else("New Tab");
```

Reply:
Sadly that doesn't compile, the ownership is a bit tricky here since we need a &str, but we have to create the underlying String from the ServoUrl first. I tried rewriting it a bit but the match seems like the simplest solution so far.